lib/spack/spack/__init__.py

```diff
@@ -121,16 +121,27 @@
 #-----------------------------------------------------------------------------
 _config = spack.config.get_config('config')
 
+# Check if user_config paths are enabled, add it to the config scope if it is.
+if _config.get('user_config', True):
+    user_config_path = os.path.expanduser('~/.spack/')
+    spack.config.ConfigScope('user', user_config_path)
+    _platform = spack.architecture.platform().name
+    spack.config.ConfigScope('user/%s' % _platform, os.path.join(user_config_path, _platform))
 
 # Path where downloaded source code is cached
 cache_path = canonicalize_path(
     _config.get('source_cache', join_path(var_path, "cache")))
 fetch_cache = spack.fetch_strategy.FsCache(cache_path)
 
 
-# cache for miscellaneous stuff.
+# Cache for miscellaneous stuff. It defaults to being within the user config directory,
+# Otherwise it ends up next to the regular cache.
+_default_misc_cache_path = os.path.join(os.path.dirname(cache_path), 'misc_cache')
+if user_config_path is not None:
+    _default_misc_cache_path = os.path.join(user_config_path, 'cache')
+
 misc_cache_path = canonicalize_path(
-    _config.get('misc_cache', join_path(user_config_path, 'cache')))
+    _config.get('misc_cache', _default_misc_cache_path))
```

Comment:
I don't know enough about the internal workings of spack, but this may interfere with multiple people trying to access it at the same time. I'm not really sure which kind of information is stored in the misc-cache anyways, though...

Reply:
misc_cache is a FileCache object; all the read and writes are done with a file lock. It should be fine. 

My main concern in this regard is permissions, but that's a general problem with a shared spack install anyway. The easy answer is for all users to use a shared group, and have the group sticky bit set on the root directory.